src/clj_ldap/client.clj

```diff
@@ -200,16 +200,16 @@
 
 (defn- connect-to-host
   "Connect to a single host"
-  [{:keys [num-connections startTLS?]
+  [{:keys [num-connections max-connections startTLS?]
```

Comment:
Hi @yuanbohan I think we can also add initial-connections to make the distinction more clear. I just tested the following and it seems to work. What do you think?
[{:keys [num-connections initial-connections max-connections startTLS?]
    :as options
    :or {num-connections 1 initial-connections num-connections
         max-connections initial-connections startTLS? false}}]

Reply:
ok

Reply:
`num-connections` may be renamed `init-connections`.

Reply:
It's better to support old version. Users can migrate to newer version without changing any configuration.

Reply:
We cannot rename num-connections unfortunately. But we can introduce both initial-connections and max-connections. This gives the caller several choices:

Use the current num-connections with no difference:
```
(ldap/connect {:host {:address "localhost" :port port}
                         :num-connections 4})
```

Use just max-connections, which would imply initial-connections = 1.
```
(ldap/connect {:host {:address "localhost" :port port}
                         :max-connections 4})
```

Use both initial and max-connections:
```
(ldap/connect {:host {:address "localhost" :port port}
                         :initial-connections 4 :max-connections 8})
```

All of the above works with the code adjustment I posted above.